main.go

```diff
@@ -55,8 +79,16 @@ func main() {
 	var wg sync.WaitGroup
 
 	// 4. Собираем числа из каналов outs
-	// ...
-
+	for i, out := range outs {
+		wg.Add(1)
+		go func(in <-chan int64, i int64) {
```

Comment:
Лучше использовать другие имена

```suggestion
		go func(in <-chan int64, j int64) {
```